scripts/src/utils/utils.ts

```diff
@@ -69,14 +71,21 @@ export enum IdPrefix {
 const ID_LENGTH = 20;
 const ID_CHARS = "1234567890abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
 
+/**
+ * Generating id uses cryptographic randomness
+ * randomInts is an array filled with random integers 0-255
+ * Every int converted into ID_CHARS place
+ *
+ * @param      {IdPrefix}  prefix  The id prefix
+ * @return     {string}  random string of ID_LENGTH length
+ */
 export function generateId(prefix: IdPrefix | string = IdPrefix.None): string {
-	let id = "";
-
-	while (id.length < ID_LENGTH) {
-		id += ID_CHARS[randomInteger(0, ID_CHARS.length)];
-	}
+	const buffer = Buffer.alloc(ID_LENGTH);
```

Comment:
interesting.
i thought that making sure we don't override the user on the db is enough, but changing how we generate ids can also work.
i've never used this method, is it optimal? we generate a lot of random ids...

Reply:
Sure, it's as optimal as generating ID_LENGTH true random numbers can be, plus it uses threadpool under the hood
`randomInteger` function that is much more used I didn't touch